src/js/component.js

```diff
@@ -1425,16 +1425,20 @@ class Component {
    * Registers a component
    *
    * @param {String} name Name of the component to register
-   * @param {Object} comp The component to register
+   * @param {Object} Comp The component to register
    * @static
    * @method registerComponent
    */
-  static registerComponent(name, comp) {
-    if (!name) {
-      return;
+  static registerComponent(name, Comp) {
+    if (typeof name !== 'string' || !(/\S/).test(name)) {
```

Comment:
will this match empty string? EDIT: it may be a good idea just to comment what that regex matches